bin/runtime/src/lib.rs

```diff
@@ -1015,10 +1015,35 @@ impl_runtime_apis! {
 
 #[cfg(test)]
 mod tests {
-    use crate::VERSION;
+    use super::*;
+    use primitives::HEAP_PAGES;
+    use smallvec::Array;
+    use frame_support::traits::Get;
+
 
     #[test]
     fn state_version_must_be_zero() {
         assert_eq!(0, VERSION.state_version);
     }
+
+    #[test]
+    fn check_contracts_memory_parameters() {
+        // Memory limit of one instance of a runtime 
+        const MAX_RUNTIME_MEM: u32 = HEAP_PAGES as u32 * 64 * 1024;
+        // Max stack size defined by wasmi - 1MB
+        const MAX_STACK_SIZE: u32 = 1024 * 1024;
+        // Max heap size is 16 mempages of 64KB each - 1MB
+        let max_heap_size = <Runtime as pallet_contracts::Config>::Schedule::get()
+            .limits
+            .max_memory_size();
+        // Max call depth is CallStack::size() + 1
+        let max_call_depth = <Runtime as pallet_contracts::Config>::CallStack::size() as u32 + 1;
+        // Max code len
+        let max_code_len: u32 = <Runtime as pallet_contracts::Config>::MaxCodeLen::get();
+
+        let lhs = max_call_depth * (72 * max_code_len + max_heap_size + MAX_STACK_SIZE);
+        let rhs = MAX_RUNTIME_MEM * 3 / 4;
```

Comment:
Could you write a one-line comment where this 3/4 is coming from, and that this is not actually enforced in the runtime :P

Reply:
I was hesitating whether I should ask the same question or I will come across as 🙈 🙈 🙉

Reply:
Done.